Sources/LanguageServerProtocolJSONRPC/DisableSigpipe.swift

```diff
@@ -0,0 +1,33 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2021 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+
+#if canImport(Glibc)
+import Glibc
+#endif
+
+#if os(Linux) || os(Android)
+// This is a lazily initialised global variable that when read for the first time, will ignore SIGPIPE.
+private let globallyIgnoredSIGPIPE: Bool = {
```

Comment:
This feels overcomplicated.  Just put this as a static variable inside JSONRPCConnection and access it from the initializer.  We don't need a public API for this.

Reply:
My line of thinking was that since this is a global property of the process, it should also be toggled by a global function. And I don’t think this really belongs in `JSONRPCConnection` either. `JSONRCPConnection` just requests `SIGPIPE` to be ignored, it shouldn’t need to worry single dispatch etc.

(also: I just copy-pased this code as-is from `swift-nio`)

Reply:
Let's make it an internal function at least.

Reply:
Sounds like a good compromise 👍